twitch4j/src/main/java/com/github/twitch4j/TwitchClientHelper.java

```diff
@@ -308,50 +331,49 @@ public TwitchClientHelper(TwitchHelix twitchHelix, EventManager eventManager, Sc
+
+            if (startedAt == null) {
+                // initialize clip query window started_at
+                nextRequestCanBeImmediate = windowStart.compareAndSet(null, now);
+            } else {
+                // get all clips in range [startedAt, now]
+                final List<Clip> clips = getClips(channelId, startedAt, now);
 
-    /**
-     * Enable StreamEvent Listener
-     *
-     * @param channelName Channel Name
-     */
-    @Nullable
-    public User enableStreamEventListener(String channelName) {
-        UserList users = twitchHelix.getUsers(null, null, Collections.singletonList(channelName)).execute();
-
-        if (users.getUsers().size() == 1) {
-            User user = users.getUsers().get(0);
-            if (enableStreamEventListener(user.getId(), user.getLogin()))
-                return user;
-        } else {
-            log.error("Failed to add channel {} to stream event listener!", channelName);
-        }
+                // cache channel name if unknown and construct event channel to be passed to events
+                if (!clips.isEmpty() && clips.get(0) != null && currentChannelCache.getUserName() == null) {
+                    currentChannelCache.setUserName(clips.get(0).getBroadcasterName());
+                }
+                final EventChannel channel = new EventChannel(channelId, currentChannelCache.getUserName());
 
-        return null;
-    }
+                // loop through queried clips
+                Instant maxCreatedAt = startedAt;
+                for (Clip clip : clips) {
+                    if (clip != null && clip.getCreatedAtInstant() != null && clip.getCreatedAtInstant().compareTo(startedAt) > 0) {
+                        eventManager.publish(new ChannelClipCreatedEvent(channel, clip)); // found a new clip
 
-    /**
-     * Enable StreamEvent Listener for the given channel names
-     *
-     * @param channelNames the channel names to be added
-     */
-    public Collection<User> enableStreamEventListener(Iterable<String> channelNames) {
-        return CollectionUtils.chunked(channelNames, MAX_LIMIT).stream()
-            .map(channels -> twitchHelix.getUsers(null, null, channels).execute())
-            .map(UserList::getUsers)
-            .filter(Objects::nonNull)
-            .flatMap(Collection::stream)
-            .filter(user -> enableStreamEventListener(user.getId(), user.getLogin()))
-            .collect(Collectors.toList());
+                        if (clip.getCreatedAtInstant().compareTo(maxCreatedAt) > 0)
+                            maxCreatedAt = clip.getCreatedAtInstant(); // keep track of most recently created clip
+                    }
+                }
+
+                // next clip window should start just after the most recent clip we've seen for this channel
+                final Instant nextStartedAt = maxCreatedAt;
+                windowStart.updateAndGet(old -> old == null || old.compareTo(nextStartedAt) < 0 ? nextStartedAt : old);
```

Comment:
this relies on the assumption that once clips for a certain timestamp, `t*`, are available in the API, later calls to the API won't yield new clips that have `t <= t*`